### PR TITLE
feat: add route for repos at org level

### PR DESCRIPTION
**Pull request by JayCeeJr**
Related Issue: https://github.com/go-vela/community/issues/332

Add route to get a list of repos enabled for an org: /api/v1/repos/::orgName

Move route to get a list a builds at an org level (this route appears to be unused by the UI, I plan to use it for the UI portion outlined in the issue above):
From api/v1/repos/::orgName
To: api/v1/repos/::orgName/builds

### CODE REVIEW

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:400]:
```suggestion
//     description: Unable to retrieve the org
```

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:404]:
```suggestion
//     description: Unable to retrieve the org
```

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:416]:
```suggestion
	logrus.Infof("Reading repos for org %s", org)
```

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:473]: Seems like this should be implemented.

File: database/postgres/repo_count.go
Comment (JordanSussman) [inline database/postgres/repo_count.go:33]:
```suggestion
	logrus.Tracef("getting count of repos for org %s in the database", org)
```

File: database/sqlite/dml/repo.go
Comment (JordanSussman) [inline database/sqlite/dml/repo.go:47]:
```suggestion
	// the count of repos for a org in the database.
```

**Review by JordanSussman — CHANGES_REQUESTED** [review]
initial feedback

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:454]:
```suggestion
	// send API call to capture the list of repos for the org
```

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:457]:
```suggestion
		retErr := fmt.Errorf("unable to get repos for org %s: %w", u.GetName(), err)
```

**Review by JordanSussman — CHANGES_REQUESTED** [review]
more feedback

File: api/repo.go
Comment (JordanSussman) [inline api/repo.go:455]:
```suggestion
	r, err := database.FromContext(c).GetOrgRepoList(u, page, perPage)
```

File: api/repo.go
Comment (JayCeeJr) [reply api/repo.go]: Added filtering for both repos and builds based on private repo flag

File: api/build.go
Comment (JordanSussman) [inline api/build.go:525]: Should this return the error instead of moving on?

File: database/postgres/repo_list.go
Comment (JordanSussman) [inline database/postgres/repo_list.go:60]:
```suggestion
	logrus.Tracef("getting private repos for org %s from the database", org)
```

File: api/build.go
Comment (JordanSussman) [inline api/build.go:533]: Should this return the error instead of moving on?

**Review by JordanSussman — COMMENTED** [review]
more input

File: api/build.go
Comment (JayCeeJr) [reply api/build.go:525]: I'll have it fail here, since the DB would most likely be the issue and it will fail on the next call as well.

File: api/build.go
Comment (JayCeeJr) [reply api/build.go:533]: If the github api is down or rate limited this will fail safe and hide the private repos. I don't think it would be ideal to fail the whole process because of a github issue.

**Comment by wass3r** [comment]
looks like the merge of https://github.com/go-vela/server/pull/444 created some conflicts - sorry :(

File: api/repo.go
Comment (wass3r) [inline api/repo.go:481]: do we actually need an blank entry? could we instead of `var excludeList []string` do `excludeList := []string{}` and avoid this?

File: api/repo.go
Comment (wass3r) [inline api/repo.go:459]: From here to the below loop and such, can we add some more comments about why we're doing this?

File: database/postgres/repo_count_test.go
Comment (wass3r) [inline database/postgres/repo_count_test.go:166]: I may have missed it but I think we want a test for the exclude functionality as well, right?

File: database/postgres/repo_list_test.go
Comment (wass3r) [inline database/postgres/repo_list_test.go:151]: similar to previous comment, we want a test where the exclude is populated as well, right?

File: database/postgres/build_list_test.go
Comment (wass3r) [inline database/postgres/build_list_test.go:146]: include a test with populated exclusion list?

File: database/postgres/build_count_test.go
Comment (wass3r) [inline database/postgres/build_count_test.go:193]: include a test with populated exclusion list?

**Review by wass3r — CHANGES_REQUESTED** [review]
thanks for the work/PR.. have few requested changes. let me know if you have questions.

File: api/repo.go
Comment (JayCeeJr) [reply api/repo.go:481]: I did that initially, and the query didn't return any results. The empty string will just add an empty string in the where clause. 
```sql
SELECT ...
WHERE name NOT IN ("")
```
And there will never be a repo with an empty string as a name.

File: api/repo.go
Comment (JayCeeJr) [reply api/repo.go:467]: I think we could do this, it will add another api call to github, for non-admins. however reduce the number of calls to one for admins.

File: api/repo.go
Comment (wass3r) [reply api/repo.go:490]: OK thanks, that's what I feared. Can we add a comment above the code block that adds the empty entry to call that out more?

File: api/repo.go
Comment (JayCeeJr) [reply api/repo.go]: True, Not sure how we would address that. I believe the perm call is a middleware function, so we do have some control over what is returned.

Comment (wass3r) [reply api/repo.go]: well, the middleware calls `source.FromContext(c).` methods. I guess I was thinking we might have to add to the service interface (see `source/service.go` and implement it for `github`.

Comment (wass3r) [reply api/repo.go]: @jbrockopp thoughts?

File: database/postgres/repo_list_test.go
Comment (JayCeeJr) [reply database/postgres/repo_list_test.go:151]: I'll add a test, however we are mocking the result-set, so not sure it accomplishes anything.

Any plans on setting up a Postgres instance for tests?

Comment (wass3r) [reply database/postgres/repo_list_test.go:151]: Not sure I follow. We create a test database and seed it with data and then mock the response we expect to get given the test.

File: api/build.go
Comment (jbrockopp) [inline api/build.go:567]: I think we're missing a couple parameters to provide to this function specifically around pagination.

```suggestion
		privateRepos, err := database.FromContext(c).GetOrgPrivateRepoList(o, page, perPage)
```

File: api/repo.go
Comment (jbrockopp) [inline api/repo.go:466]: I think we're missing a couple parameters to provide to this function specifically around pagination.

```suggestion
		privateRepos, err := database.FromContext(c).GetOrgPrivateRepoList(org, page, perPage)
```

File: database/postgres/build_list.go
Comment (jbrockopp) [inline database/postgres/build_list.go:178]: I don't think we want to add this function as it was removed with:

https://github.com/go-vela/server/pull/443

File: database/postgres/dml/build.go
Comment (jbrockopp) [inline database/postgres/dml/build.go:109]: I don't think we want to add this query as it was removed with:

https://github.com/go-vela/server/pull/443

File: database/postgres/dml/repo.go
Comment (jbrockopp) [inline database/postgres/dml/repo.go:33]: I think we're missing `LIMIT` and `OFFSET` from this query.

AFAICT, this would capture every repo in the given org with a visibility of `private`.

This can cause a problem if the number of repos returned by the query is abnoramlly large.

You can use the below for reference:

https://github.com/go-vela/server/blob/fefd9eec0fbad52a17717104b23375a0bcf075e9/database/postgres/dml/repo.go#L15-L23

File: database/postgres/repo_list.go
Comment (jbrockopp) [inline database/postgres/repo_list.go:59]: Think we're missing a couple parameters here for pagination (`page` and `perPage`):

```suggestion
func (c *client) GetOrgPrivateRepoList(org string, page, perPage int) ([]*library.Repo, error) {
```

File: database/sqlite/build_list.go
Comment (jbrockopp) [inline database/sqlite/build_list.go:179]: I don't think we want to add this function as it was removed with:

https://github.com/go-vela/server/pull/443

File: database/sqlite/dml/build.go
Comment (jbrockopp) [inline database/sqlite/dml/build.go:126]: I don't think we want to add this query as it was removed with:

https://github.com/go-vela/server/pull/443

File: database/sqlite/dml/repo.go
Comment (jbrockopp) [inline database/sqlite/dml/repo.go:32]: I think we're missing `LIMIT` and `OFFSET` from this query.

AFAICT, this would capture every repo in the given org with a visibility of `private`.

This can cause a problem if the number of repos returned by the query is abnoramlly large.

You can use the below for reference:

https://github.com/go-vela/server/blob/fefd9eec0fbad52a17717104b23375a0bcf075e9/database/sqlite/dml/repo.go#L15-L23

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: api/build.go
Comment (JayCeeJr) [reply api/build.go:567]: @jbrockopp @wass3r 

After thinking about this for a while, maybe we just want to exclude builds and repos that are private from the Org pages. 
We took this approach to avoid data leakage, however I think the overhead of checking permissions by hitting the github api will not scale well.

Comment (jbrockopp) [reply api/build.go:567]: @JayCeeJr I'm not opposed to that approach and I think that's a fair way to handle it 👍 

We can always explore attempting to add this functionality in at a later date if it's found valuable by our users.

And perhaps in the future we'll be storing user permissions for orgs and repos in our database?

Thinking that maybe it would help resolve some concerns with taxing the GitHub API.

File: database/postgres/repo_list_test.go
Comment (wass3r) [reply database/postgres/repo_list_test.go:151]: that's fair. i was thinking about sqlite, i think :)

File: api/build.go
Comment (JayCeeJr) [reply api/build.go]: @jbrockopp 
I took a middle-ground approach to this. I check if the user is an admin, if not i only show public repos for the org. Same logic applies to the builds.

**Review by wass3r — APPROVED** [review]
thanks a ton for working through the changes 👍🏼

**Review by jbrockopp — APPROVED** [review]
LGTM

**Comment by jbrockopp** [comment]
@JordanSussman we'll plan to merge this PR once it has your approval since you had requested feedback 👍

**Review by JordanSussman — APPROVED** [review]

**Comment by JordanSussman** [comment]
> @JordanSussman we'll plan to merge this PR once it has your approval since you had requested feedback 👍

Approved!

File: api/build.go
Comment (JordanSussman) [reply api/build.go]: May need to utilize the [list organization repos](https://docs.github.com/en/rest/reference/repos#list-organization-repositories) endpoint or [graphql](https://github.com/shurcooL/githubv4) in the future.

**Pull request merged**
